pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -657,6 +662,40 @@ func (r *ReconcilePerconaServerMongoDB) ensureSecurityKey(cr *api.PerconaServerM
 	return created, nil
 }
 
+func (r *ReconcilePerconaServerMongoDB) deleteOrphanPVCs(cr *api.PerconaServerMongoDB) error {
+	for _, f := range cr.GetFinalizers() {
+		switch f {
+		case "delete-psmdb-pvc":
+			// remove orphan pvc
+			mongodPVCs, err := r.getMongodPVCs(cr)
+			if err != nil {
+				return err
+			}
+			mongodPods, err := r.getMongodPods(cr)
+			if err != nil {
+				return err
+			}
```

Comment:
We need to delete PVCs only on the downscale when users reduce the replicaSet option `spec.replsets[].size` inside CR.
We think pod data should be preserved during Pod restart.